sdk/CONTRIBUTING.md

```diff
@@ -0,0 +1,86 @@
+# SDK Contribution Guide
+
+This document serves as a guide for creating a new Dagger SDK, but also to help maintain consistency between existing ones.
+
+## Purpose of an SDK
+
+- Best DX
+- Technically, any language that can make HTTP requests, can talk to Dagger.
+  - However, without concurrency features there’s a bottleneck on performance that makes the chosen language very limiting.
+- Provide the best developer experience for working with Dagger, in a way that feels natural to the users of the chosen language (idiomatic).
+
+## Codegen
```

Comment:
I'm unclear about what this section represents. Some of the bulleted items refer to tools while some are prescriptive in nature and some refer to concepts. Maybe classify further/reclassify and/or add notes